var/spack/repos/builtin/packages/py-breakseq2/package.py

```diff
@@ -0,0 +1,37 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyBreakseq2(PythonPackage):
+    """nucleotide-resolution analysis of structural variants"""
+
+    homepage = "http://bioinform.github.io/breakseq2/"
+    url      = "https://github.com/bioinform/breakseq2/archive/2.2.tar.gz"
+
+    version('2.2', '6fd5a103c2781717b0b1d0efcbdc17e7')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-biopython',        type=('build', 'run'))
```

Comment:
Can you remove the extra spaces here? Also, you're missing some dependencies:
```python
install_requires = ["cython", "pysam==0.7.7", "biopython==1.65"],         
```
Unfortunately, you're going to have to be just as explicit about the version requirements, otherwise Python will fetch them from PyPI and build them outside of Spack. This may work for you, but not for another user on an air-gapped network.

Reply:
Done.

Reply:
Just for my my education:  How did you know specific versions of pysam and biopython were required? There wasnt anything about that in the build instructions.

Reply:
There are a bunch of places where python-packages can put their dependencies, this one puts it in `setup.py` (https://github.com/bioinform/breakseq2/blob/master/setup.py)

There are also packages that put it in files like `requirements.txt` or so

Reply:
P.S. The dependencies in `requirements.txt` are not always reliable, as they often contain things needed to build the documentation or run unit tests. The ones specified in `setup.py` are more reliable, as they are used by `pip` during installation. So if you're a Python developer and you want your software to install correctly, you must specify it in your `setup.py`. Of course, this feature is only available for developers who use `setuptools`, so those who use `distutils` often find other ways to check for dependencies (attempt to import and crash if it can't find a dependency).